src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -251,11 +249,12 @@ internal void StartMessageBatchReceiver(CancellationToken cancellationToken)
                             return;
                         }
 
-                        if (_isSessionsEnabled)
+                        if (_isSessionsEnabled && ( receiver == null || receiver.IsClosedOrClosing))
                         {
                             try
                             {
                                 receiver = await sessionClient.AcceptMessageSessionAsync();
+                                receiver.PrefetchCount = _serviceBusOptions.PrefetchCount;
                             }
                             catch (ServiceBusTimeoutException)
                             {

```

Comment:
Would it be useful to have a debug statement here when we there are not sessions to accept or too much of a perf. hit?

Reply:
I think there can be too much noise in the logs